src/data/real/nnreal.lean

```diff
@@ -280,13 +304,26 @@ begin
   { assume h, rw [h] }
 end
 
+lemma of_real_mul {p q : ℝ} (hp : 0 ≤ p) (hq : 0 ≤ q) :
+  nnreal.of_real (p * q) = nnreal.of_real p * nnreal.of_real q :=
+nnreal.eq $
+by { have := max_eq_left (zero_le_mul hp hq), simpa [nnreal.of_real, hp, hq, max_eq_left] }
+
 end mul
 
 section sub
 
 lemma sub_eq_zero {r p : nnreal} (h : r ≤ p) : r - p = 0 :=
 nnreal.eq $ max_eq_right $ sub_le_iff_le_add.2 $ by simpa [nnreal.coe_le] using h
 
+protected lemma sub_lt_self {r p : nnreal} : r > 0 → p > 0 → r - p < r :=
```

Comment:
The inequalities should always be written as `0 < r` instead of `r > 0`, for rewrites to work properly everywhere.